File: ProgAsynchrone-ChatApp/MultiCLients/C#/Server/ClientHandler.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading.Tasks;

namespace Server
{
    internal class ClientHandler
    {
        private readonly TcpClient _client;

        public ClientHandler(TcpClient client)
        {
            _client = client;
        }

        public async Task HandleClientMessagesAsync()
        {
            using (var stream = _client.GetStream())
            {
                using (var reader = new StreamReader(stream))
                {
                    StreamWriter writer = new StreamWriter(stream) { AutoFlush = true };

                    // Read user name
                    await writer.WriteLineAsync("Enter your name: ");
                    string userName = await reader.ReadLineAsync();
                    Console.WriteLine(userName + " has joined the chat.");

                    try
                    {
                        string message;
                        while ((message = await reader.ReadLineAsync()) != null)
                        {
                            Console.WriteLine(userName + ": " + message);
                        }
                    }
                    catch (IOException)
                    {
                        // Handle disconnection
                    }
                    finally
                    {
                        Console.WriteLine($"{userName} has left the chat.");
                    }
                }
            };
        }

    }
}

File: ProgAsynchrone-ChatApp/MultiCLients/Java/Server/src/ClientHandler.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.PrintWriter;
import java.net.Socket;

public class ClientHandler implements Runnable {
    private final Socket clientSocket;

    public ClientHandler(Socket clientSocket) {
        this.clientSocket = clientSocket;
    }

    @Override
    public void run() {
        try (var stream = clientSocket.getInputStream();
             var reader = new BufferedReader(new InputStreamReader(stream));
             var writer = new PrintWriter(new OutputStreamWriter(clientSocket.getOutputStream()), true)) {

            // Read user name
            writer.println("Enter your name: ");
            String userName = reader.readLine();
            System.out.println(userName + " has joined the chat.");

            try {
                String message;
                while ((message = reader.readLine()) != null) {
                    System.out.println(userName + ": " + message);
                }
            } catch (IOException e) {
                // Handle disconnection
            } finally {
                System.out.println(userName + " has left the chat.");
            }
        } catch (IOException e) {
            e.printStackTrace();
        }
    }
}